src/maven-repair/pom.xml

```diff
@@ -146,6 +146,21 @@
             <artifactId>repairnator-core</artifactId>
             <version>3.3-SNAPSHOT</version>
         </dependency>
+        <dependency>
+            <groupId>fr.inria.gforge.spoon</groupId>
+            <artifactId>spoon-core</artifactId>
+            <version>7.0.0</version>
+        </dependency>
+        <dependency>
+            <groupId>commons-cli</groupId>
+            <artifactId>commons-cli</artifactId>
+            <version>1.3</version>
+        </dependency>
+        <dependency>
+            <groupId>org.json</groupId>
+            <artifactId>json</artifactId>
+            <version>20160810</version>
+        </dependency>
```

Comment:
I'm not sure this is the best approach, albeit it works. Shouldn't the dependencies be fetched transitively?

Reply:
> I'm not sure this is the best approach

I agree with you. At least in the case of `spoon-core` I could see that the dependency metadata (pom and sha files) is downloaded, but unless I put these here explicitly it does not download the jar. I assumed the same for `commons-cli` and `json`.

Also, I checked and the maven-repair tests work locally in my pc only because those jars were already there as a result of compiling another module.

If there is a better way to trigger the jar download, we can replace this.

Reply:
Yes, it's very weird behavior indeed. While debugging I also saw that pom and sha files were downloaded, just not the jar file.

I think it's important to get this right tho, because having sub-dependencies hardcoded makes it hard to maintain. 

I will look further into it after my exam tomorrow.

Reply:
`mvn dependency:tree` for `maven-repair` is in https://gist.github.com/andre15silva/254c0de8ef2d2a7ac45c834f404119c4

The issue is that these three dependencies are being omitted because of conflicts with sub-dependencies from other repair tools that use these libraries but with a different version.

```
[INFO]    +- (fr.inria.gforge.spoon:spoon-core:jar:7.0.0:compile - omitted for conflict with 6.0.0)
[INFO]    +- (commons-cli:commons-cli:jar:1.3:compile - omitted for conflict with 1.2)
[INFO]    \- (org.json:json:jar:20160810:compile - omitted for conflict with 20160212)
```

If we added the `Nopol` dependency first, it would fetch these versions. It's also not a very elegant solution tho. Yours is probably better, as it is explicitly written.

Another option would be to change `getClassPathFromPom` to take into account that the version explicited in the repair tools pom files may not be available locally and that another version has to be used. I'm not sure this is a very good idea tho.

Reply:
> If we added the Nopol dependency first, it would fetch these versions. It's also not a very elegant solution tho. Yours is probably better, as it is explicitly written.

I believe I tried this, and it did not resolve the conflicts.

> Another option would be to change getClassPathFromPom to take into account that the version explicited in the repair tools pom files may not be available locally and that another version has to be used.

I don't think this is a good idea either, we may create an even more complicated compatibility/dependency issue.